### PR TITLE
Orders backend task 2020 00058

### PR DESCRIPTION
**Pull request by NeerajSharma2020**
Pull request for Order.

### CODE REVIEW

File: src/main/java/com/st/ims/utility/AppUtility.java [hunk 1]
```diff
@@ -3,72 +3,133 @@
...
+
 	/*
-	 * This method will take invoice amount and commission percentage and will return
+	 * This method will take order amount and commission percentage and will return
 	 * commission amount.
 	 */
-	private static double getCommissionAmount(double invoiceAmount , double commissionPercentage) {
-		return invoiceAmount*(commissionPercentage/100);
+	private static double getCommissionAmount(double invoiceAmount, double commissionPercentage) {
+		return invoiceAmount * (commissionPercentage / 100);
```
Comment (premshani83) [inline src/main/java/com/st/ims/utility/AppUtility.java]: please define constant for 100

File: src/main/java/com/st/ims/utility/AppUtility.java [hunk 2]
```diff
@@ -3,72 +3,133 @@
...
+		} catch (NoSuchAlgorithmException e) {
+			logger.error("Exception while configuring dueDate,orderNumber and commissionAmount.", e);
+		}
+		return null;
+	}
+	
+	/* This method will return OrderDetails List with products. */
+	public static List<OrderDetails> getOrderDetailsList(Order order, List<OrderProducts> orderProductList){
+		try {
+			List<OrderDetails> orderDetailsList = new ArrayList<>();
```
Comment (premshani83) [inline src/main/java/com/st/ims/utility/AppUtility.java]: should decleare with null

File: src/main/java/com/st/ims/utility/AppUtility.java [hunk 3]
```diff
@@ -3,72 +3,133 @@
...
+		} catch (Exception e) {
+			logger.error("Exception while getting OrderDetailsList.",e);
+		}
+		return Collections.emptyList();
+	}
+	
+	/* This method will create new OrderDetails list for update operation. */
+	public static List<OrderDetails> getUpdatedOrderList(Order order,List<OrderProducts> orderProductList){
+		try {
+			List<OrderDetails> orderDetailsList = new ArrayList<>();
```
Comment (premshani83) [inline src/main/java/com/st/ims/utility/AppUtility.java]: should decleare with null

**Review by premshani83 — COMMENTED** [review]
please change

File: src/main/java/com/st/ims/utility/AppUtility.java (hunk 3 above)
Comment (NeerajSharma2020) [reply src/main/java/com/st/ims/utility/AppUtility.java]: Compiler doesn,t allows to be null at this location.

File: src/main/java/com/st/ims/utility/AppUtility.java (hunk 2 above)
Comment (NeerajSharma2020) [reply src/main/java/com/st/ims/utility/AppUtility.java]: Compiler doesn,t allows to be null at this location.

File: src/main/java/com/st/ims/utility/AppUtility.java (hunk 1 above)
Comment (NeerajSharma2020) [reply src/main/java/com/st/ims/utility/AppUtility.java]: done

**Comment by premshani83** [comment]
please resolve conflict

Conflicting files
src/main/java/com/st/ims/utility/AppUtility.java

File: src/main/java/com/st/ims/resolvers/OrderDetailsQueryResolver.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	 * list of all OrderDetials object for that if exist in database.
+	 */
+	public List<OrderDetails> findOrderDetailById(int orderId) {
+		try {
+		Order order = orderRepo.findById(orderId).orElse(null); 
+		if(order != null) {
+	    	 return order.getOrderDetails();
+	   
+	     }else {
+	    	 logger.error("Error while while getting list for orderId, may be order with id "+orderId+" doesn,t exists.");
```
Comment (premshani83) [inline src/main/java/com/st/ims/resolvers/OrderDetailsQueryResolver.java]: this is not error, it should be info 

logger.error("Error while while getting list for orderId, may be order with id "+orderId+" doesn,t exists.");

**Review by premshani83 — COMMENTED** [review]
please change

**Review by premshani83 — APPROVED** [review]
merge into develop

**Comment by NeerajSharma2020** [comment]
conflicts.

**Pull request closed without merging**